pg_shard.c

```diff
@@ -625,8 +664,7 @@ PlanSequentialScan(Query *query, int cursorOptions, ParamListInfo boundParams)
 	enable_indexscan = false;
 	enable_bitmapscan = false;
 
-	queryCopy = copyObject(query);
-	sequentialScanPlan = standard_planner(queryCopy, cursorOptions, boundParams);
+	sequentialScanPlan = standard_planner(query, cursorOptions, boundParams);
```

Comment:
Hmm, so now since we modify the localQuery, it makes sense to make a copy of it before we update the quals to NIL. Lets change the function comment here to document that it modifies the inputs. On looking at it a bit more, I like that the planner modifications to the localQuery which generated the plannedStmt are available to the caller of this function.